### ISSUE TITLE
Enable multiple templates to be assigned to a single host

### ISSUE THREAD

**Issue by TheNetworkGuy** [TheNetworkGuy/netbox-zabbix-sync#27]
Due to the development in https://support.zabbix.com/browse/ZBXNEXT-8087 it is most likely that a single template will not suffice to monitor all data from a host.

I would like to propose to move the template name from a custom field to config context. This will result in the following improvements:

- Allows a list-type with multiple template names to be assigned to a single host or device type.
- Allow users to overwrite default template assignments since the config context of a host is a higher level.
- Allows possible future developments of integrating VM support since config context can be applied to different types of objects, including devices and VM's.

This would be a major change in the way of administrating and linking Netbox devices with a template. Because of this i would like to make a migration script that takes the current values and sets them to their correct config context value.

Would love to hear your thoughts on this.

**Comment by Eliesmbr** [comment]
Since we already have everything in netbox regarding host configuration in the config context, I would be very happy about this change. We don't use zabbix yet, but are considering using it and how we want to use it with netbox as our source of truth. We have always avoided custom fields as there is no inheritance of config based on tags, site or roles.

**Comment by TheNetworkGuy** [comment]
FYI still working on this and looking to implement this. Even though the original plans in ZBXNEXT-8087 are discontinued. However its a lot of code that needs to be rewritten including the ConsistencyCheck() function.

TLDR; not forgotten but need to find time to implement this.

**Comment by TheNetworkGuy** [comment]
@Eliesmbr I've implemented this feature in the new version of the script. Could you take a look at it and check if this fulfills your requirements?

**Comment by Eliesmbr** [comment]
Thank you for your work, but in the meantime i create my own netbox-zabbix sync script which works a bit different. When i have time, i will upload this to Github.

**Comment by TheNetworkGuy** [comment]
Glad to hear. In that case i'll close this issue.
